crates/fuel-core/src/service/adapters/graphql_api.rs

```diff
@@ -175,3 +204,62 @@ impl DatabaseChain for Database {
 }
 
 impl DatabasePort for Database {}
+
+impl InsertTx for TxPoolService {
+    fn insert(&self, txs: Vec<Arc<Transaction>>) -> Vec<anyhow::Result<InsertionResult>> {
+        self.shared.insert(txs)
+    }
+}
+
+impl TxSubscription for TxPoolService {
+    fn tx_update_subscribe(
+        &self,
+    ) -> fuel_core_services::stream::BoxStream<
+        Result<fuel_core_txpool::service::TxUpdate, BroadcastStreamRecvError>,
```

Comment:
One thing to keep in mind here is that you can get [lagged errors](https://docs.rs/tokio/latest/tokio/sync/broadcast/error/enum.RecvError.html#variant.Lagged) which you probably don't want to pass on to the consumer. You could filter them out and convert to an anyhow error

Reply:
This returns `BroadcastStreamRecvError`which only contains lagged out errors, so would you prefer I filter out errors entirely and just return a TxUpdate?

Reply:
Yeh I think it's best to filter out lagged errors as there's nothing the caller can really do about them

Reply:
Actually there is, looking through the upstream changes this would force [transaction_status_change](https://github.com/FuelLabs/fuel-core/blob/controlc/gql_extraction/crates/fuel-core/src/query/subscriptions.rs#L29) to accept a `TxUpdate` Non-Result (as `BroadcastStreamRecvError` only has `LaggedOut`), and `transaction_status_change` will check the [database if the status lagged out](https://github.com/FuelLabs/fuel-core/blob/controlc/gql_extraction/crates/fuel-core/src/query/subscriptions.rs#L89) to make sure the update wasn't missed

Reply:
Ah nice catch